graviti/dataframe/frame.py

```diff
@@ -168,7 +176,38 @@ def from_lazy_lists(cls: Type[_T], lazy_lists: "LazyLists") -> _T:
             The loaded :class:`~graviti.dataframe.DataFrame` object.
 
         """
-        return cls(lazy_lists)
+        # Not support loading tensorbay Dataset.
+        return cls(lazy_lists)  # type: ignore[arg-type]
+
+    @classmethod
+    def from_pyarrow(cls: Type[_T], array: pa.StructArray, schema: record) -> _T:
+        """Create DataFrame with pyarrow struct array.
+
+        Arguments:
+            array: The input pyarrow struct array.
+            schema: Schema.
+
+        Returns:
+            The loaded :class:`~graviti.dataframe.DataFrame` object.
+
+        """
+        obj: _T = object.__new__(cls)
+        obj.schema = schema
+        obj._columns = {}
+        obj._column_names = []
+
+        for pafield in array.type:
+            column_name = pafield.name
+            column_value = array.field(column_name)
+            column_schema = schema.fields[column_name]
+            obj._columns[column_name] = (
+                DataFrame.from_pyarrow(column_value, schema=column_schema)
+                if isinstance(pafield.type, pa.StructType) and isinstance(column_schema, record)
```

Comment:
why we need to check `column_schema` here?

Reply:
for mypy checking.